src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -329,6 +329,10 @@ public virtual void ValidateAsymmetricSecurityKeySize(SecurityKey key, string al
         /// <remarks>Verify is thread safe.</remarks>
         public override bool Verify(byte[] input, byte[] signature)
         {
+            // The reason this method doesn't call through to: Verify(input, 0, input.Length, signature, 0, signature.Length)
+            // Is because this method's contract is to check the entire signature, if the signature was truncated and signature.Length
+            // was passed, the signature may verify.
```

Comment:
Does this imply a truncated signature may pass verification? Wouldn't the odds of that be similar to random signature verifying (i.e. vanishingly small)?